python/plugins/livekit/composio_livekit/toolset.py

```diff
@@ -0,0 +1,212 @@
+    runtime="livekit",
+    description_char_limit=1024,
+    action_name_char_limit=64,
+):
+    """
+    Composio toolset for LiveKit integration.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def _clean_schema_properties(self, prop_value: dict) -> None:
+        """Clean individual property fields."""
+        for field in ["examples", "pattern", "default"]:
+            if field in prop_value:
+                del prop_value[field]
+
+        if "items" in prop_value and isinstance(prop_value["items"], dict):
+            if "type" not in prop_value["items"]:
+                prop_value["items"]["type"] = "string"
+
+    def _clean_schema(self, schema_obj: t.Dict[str, t.Any]) -> None:
+        """Remove examples and ensure array items have types."""
+        if "properties" in schema_obj and isinstance(schema_obj["properties"], dict):
+            for _, prop_value in schema_obj["properties"].items():
+                if isinstance(prop_value, dict):
+                    self._clean_schema_properties(prop_value)
+                    self._clean_schema(prop_value)
+
+        if "items" in schema_obj and isinstance(schema_obj["items"], dict):
+            for field in ["examples", "pattern", "default"]:
+                if field in schema_obj["items"]:
+                    del schema_obj["items"][field]
+            if "type" not in schema_obj["items"]:
+                schema_obj["items"]["type"] = "string"
+            self._clean_schema(schema_obj["items"])
+
+        for value in schema_obj.values():
+            if isinstance(value, dict):
+                self._clean_schema(value)
+            elif isinstance(value, list):
+                for item in value:
+                    if isinstance(item, dict):
+                        self._clean_schema(item)
+
+    def _wrap_tool(
+        self,
+        schema: t.Dict[str, t.Any],
+        entity_id: t.Optional[str] = None,
+    ) -> FunctionTool:
+        """Wraps composio tool as LiveKit Function tool
+
+        Args:
+            schema: The action schema to wrap
+            entity_id: Optional entity ID for executing function calls
+
+        Returns:
+            A LiveKit Function tool
+        """
+        action = schema["name"]
+        description = schema["description"]
+        schema_params = schema["parameters"]
+
+        # Ensure the schema has additionalProperties set to false
+        modified_schema = schema_params.copy()
+        modified_schema["additionalProperties"] = False
+
+        # Apply schema cleaning
+        self._clean_schema(modified_schema)
+
+        async def execute_action_wrapper(ctx: RunContext, **kwargs) -> str:
+            """Execute Composio action with the given arguments."""
+            try:
+                result = self.execute_action(
+                    action=action,
+                    params=kwargs,
+                    entity_id=entity_id or self.entity_id,
+                )
```

Comment:
The wrapper method is async but this call is blocking, use `asyncio.to_thread`
```suggestion                
                result = await asyncio.to_thread(
                    self.execute_action,
                    action=action,
                    params=kwargs,
                    entity_id=entity_id or self.entity_id,
                )
```